corecfg/corecfg.go

```diff
@@ -0,0 +1,79 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package corecfg
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/systemd"
+)
+
+var (
+	Stdout = os.Stdout
+	Stderr = os.Stderr
+)
+
+// coreSupportAvailable checks that the system has the core-support
+// interface. An error is returned if this is not the case
+func coreSupportAvailable() error {
+	_, err := systemd.SystemctlCmd("--version")
+	return err
+}
+
+func snapctlGet(key string) (string, error) {
+	raw, err := exec.Command("snapctl", "get", key).CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("Cannot run snapctl: %s", osutil.OutputErr(raw, err))
+	}
+
+	output := strings.TrimRight(string(raw), "\n")
```

Comment:
I do usually too, I was mostly cautious because a config option may contain data where the whitespace *may* have a meaning. If this is not a concern I'm happy to change it to the simpler TrimSpace.